addons/account/wizard/account_move_send.py

```diff
@@ -145,6 +139,13 @@ def _get_default_email_from(self, mail_template, lang, move):
     # COMPUTE METHODS
     # -------------------------------------------------------------------------
 
+    @api.depends('move_ids')
+    def _compute_company_id(self):
+        for wizard in self:
+            if len(wizard.move_ids.company_id) > 1:
+                raise UserError(_("You can only send from the same company."))
+            wizard.company_id = wizard.move_ids.company_id.id
```

Comment:
Why isn't it a `@api.constrains`?

Reply:
Because then it only raises on validation of the wizard and not when opening it => not user friendly